kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/maps/MapMatchersTest.kt

```diff
@@ -412,5 +412,42 @@ expected:<{
 }>"""
          }
       }
+
+
+      "matchAll" should {
+         "empty" {
+            emptyMap<String, String>() should matchAll()
+         }
+
+         "empty negated" {
+            emptyMap<String, String>() shouldNot matchAll("key" to { it shouldBe "value" })
+         }
+
```

Comment:
Perhaps adding a test for `shouldFail { emptyMap<String, String>().shouldNotMatchAll() }` would be nice, to make it clear what rules apply to empty sets and empty matchers.